A77-xds-rate-limiting-rlqs.md

```diff
@@ -0,0 +1,618 @@
+| `has(m.x)`                                         | (macro) Returns true if the map `m` has the string `"x"` as a key.                            |
+| `int`, `uint`, `double`, `string`, `bytes`, `bool` | Conversions and identities.                                                                   |
+| `==`, `!=`, `>`, `<`, `<=`, `>=`                   | Comparisons.                                                                                  |
+| `or`, `&&`, `+`, `-`, `/`, `*`, `%`, `!`           | Basic functions.                                                                              |
+
+[RE2_wiki]: https://en.wikipedia.org/wiki/RE2_(software)
+
+##### Supported CEL Variables
+
+For RLQS, only the `request` variable is supported in CEL expressions. We will
+adapt [Envoy's Request Attributes](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes#request-attributes)
+for gRPC.
+
+| Attribute           | Type                  | gRPC source                  | Envoy Description                                           |
+|---------------------|-----------------------|------------------------------|-------------------------------------------------------------|
+| `request.path`      | `string`              | Full method name<sup>1</sup> | The path portion of the URL.                                |
+| `request.url_path`  | `string`              | Same as `request.path`       | The path portion of the URL without the query string.       |
+| `request.host`      | `string`              | Authority<sup>2</sup>        | The host portion of the URL.                                |
+| `request.scheme`    | `string`              | Not set                      | The scheme portion of the URL.                              |
+| `request.method`    | `string`              | `POST`<sup>3</sup>           | Request method.                                             |
+| `request.headers`   | `map<string, string>` | `metadata`<sup>4</sup>       | All request headers indexed by the lower-cased header name. |
+| `request.referer`   | `string`              | `metadata["referer"]`        | Referer request header.                                     |
+| `request.useragent` | `string`              | `metadata["user-agent"]`     | User agent request header.                                  |
+| `request.time`      | `timestamp`           | Not set                      | Time of the first byte received.                            |
+| `request.id`        | `string`              | `metadata["x-request-id"]`   | Request ID corresponding to `x-request-id` header value     |
+| `request.protocol`  | `string`              | Not set                      | Request protocol.                                           |
+| `request.query`     | `string`              | `""`                         | The query portion of the URL.                               |
+
+###### Footnotes
+
+**<sup>1</sup> `request.path`**
```

Comment:
The language-specific details in these footnotes don't seem necessary to spell out.